### PR TITLE
Create Dockerfile to build app

### PR DESCRIPTION
**Pull request by sergiofteixeira**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+
+COPY . .
+
+RUN dep ensure
+
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags '-w' -i -o /go/bin/envoy-preflight ./main.go
+
+FROM golang:1.13.4-alpine
```
Comment (jackkleeman) [inline Dockerfile:15]:
```suggestion
FROM scratch
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+COPY --from=builder /go/bin/envoy-preflight /go/bin/envoy-preflight
```
Comment (jackkleeman) [inline Dockerfile:17]:
```suggestion
COPY --from=builder /go/bin/envoy-preflight /envoy-preflight
```

Comment (jackkleeman) [inline Dockerfile:17]: Should we add an `ENTRYPOINT` as well? So that the caller only has to provide the secondary binary as an argument? Otherwise LGTM

**Comment by jackkleeman** [comment]
Thanks for your contribution! Left one comment, but this looks great!

**Comment by AceHack** [comment]
any way we could get this for a few more distributions?  Like Debian slim?

**Comment by jackkleeman** [comment]
@AceHack what's your use case? `envoy-preflight` is just a Go binary, normally I would recommend adding it to your existing Docker images. The above docker file is handy to be able to mount in a bash script - does distro matter much if that's all we're doing? I'm open to contributions, by the way

**Comment by AceHack** [comment]
If you release the binary that would work on several different platforms that would be great.  We are using the docker file to copy the binary into our docker files in a multistage docker file.  Download the released binary into out docker would be fine too.

File: Dockerfile (hunk 2 above)
Comment (sergioteixeiraptc) [reply Dockerfile:17]: updated with entrypoint

**Comment by sergioteixeiraptc** [comment]
updated the dockerfile with an entrypoint, and yes a released binary would be usefull aswell, you could create a github action or smth to automatically release the binary.

**Comment by sergiofteixeira** [comment]
updated the dockerfile with an entrypoint, and yes a released binary would be usefull aswell, you could create a github action or smth to automatically release the binary.

**Review by jackkleeman — APPROVED** [review]

**Comment by jackkleeman** [comment]
@sergioteix @AceHack I've added a release binary https://github.com/monzo/envoy-preflight/releases/tag/v1.0, its generated by a GitHub action on new release. If there's any improvements you'd like, feel free to contribute them.

**Pull request merged**